Code/Tools/SceneAPI/SceneCore/DataTypes/Rules/IReadOnlyRule.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/RTTI/RTTI.h>
+#include <SceneAPI/SceneCore/DataTypes/Rules/IRule.h>
+
+namespace AZ
+{
+    namespace SceneAPI
+    {
+        namespace DataTypes
+        {
+            // Disables UI interaction for the node and children of the node with this rule.
+            class IReadOnlyRule : public IRule
```

Comment:
I'm not really a fan of creating a inheriting class between IRules and derived class (e.g ISkinRules) just to modify the tool tips. 
What about just add the modifyToolTips function to IRules and let the derived class handle the actual part of tool tip modification?

Reply:
I originally didn't have an IReadOnlyRule class, but it caused some serialization and linker issues due to how the systems are setup. SceneAPI / SceneCore are divided into a few pieces that make it tough to get everything linking in the right places.

Reply:
Ok I think I understand the code a little better now. It's not a "ReadOnly" Rule - instead it's basically a rule that makes whatever container holds it to become read only, correct?
Maybe there could be a better name for this? UnmodifiableRule?

Reply:
Sure, I've renamed it to that. I didn't rename it everywhere because it's using the ReadOnly system in the reflected property editor, but I renamed the rule.